homeassistant/components/hassio/__init__.py

```diff
@@ -124,6 +125,21 @@
 }
 
 
+@bind_hass
+async def async_get_addon_info(
+    hass: HomeAssistantType, slug: str, repository: str
+) -> dict:
+    """Return add-on info.
+
+    The caller of the function should handle HassioAPIError.
+    """
+    hassio = hass.data[DOMAIN]
+    addon = f"{repository}_{slug}"
+    command = f"/addons/{addon}/info"
+    result = await hassio.send_command(command, method="get")
+    return result["data"]
```

Comment:
We should use the method available on the Hassio object.

```suggestion
    return await hass.data[DOMAIN].get_addon_info(addon)
```

I think that we should also simplify this method to also take a single identifier. The fact that an add-on identifier is made up of a repository slug and ID is not something that we today expose as a feature.

Reply:
As we've discussed, there's no existing single identifier available. Concatenating repository and slug, makes it even harder to know what the identifier is. At least the slug is available in the config.json.

I'm happy to hear suggestion what the identifier should be.

Reply:
If I look at [the Supervisor API](https://github.com/home-assistant/supervisor/blob/dev/API.md), the add-on info has a `slug` field which is already prefixed with repository. That is the add-on slug. We're referencing a variable as `slug` here that is not actually that what the addon expects to be its `slug`. And none of the APIs take in the type of `slug` as we reference in this function. 

If we hardcode the OZW slug as `core_ozw` in the OZW integration, we can pass that to this function to get the info. That would work?

Reply:
The `slug` variable has been made confusing in the supervisor since it's used both for an item in the add-on config.json and for the item in the add-on info result. And the value is not the same. In the config.json the slug doesn't include the repository, while in the api info it does.

I'll concatenate them.

Reply:
The supervisor API use slug on all objects to make it available. On Supervisor with different view, a add-on slug is an ID inside repository which is unique inside the repository and the SU make an unique ID with that over all repository

Reply:
I understand how it works, but it's still confusing to name different things the same.